src/coreclr/src/jit/importer.cpp

```diff
@@ -4509,12 +4510,13 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
 
             result = HWIntrinsicInfo::lookupId(this, &sig, className, methodName, enclosingClassName);
         }
-        else if (strcmp(methodName, "get_IsSupported") == 0)
-        {
-            return NI_IsSupported_False;
-        }
-        else
+
+        if (result == NI_Illegal)
```

Comment:
This path accounts for `mustExpand` for unrecognized HWIntrinsics in particular as we know that `throw PlatformNotSupportedException` is the desired behavior for anything not recognized and which is recursive under `namespace System.Runtime.Intrinsics`.

All of the code paths are either directly recursive (`void MyMethod() => MyMethod()`) and are under one of the `S.R.I.X86` or `S.R.I.Arm` or are indirectly recursive and are directly under `S.R.I`. An indirectly recursive intrinsic is one like `Vector64.Create` which is:
```csharp
if (AdvSimd.IsSupported)
{
    return Vector64.Create(...);
}

return SoftwareFallback(...);
```

So the `throw PNSE` will be generated under the dead `AdvSimd.IsSupported` code path under minOpts and will never actually be hit.

Reply:
This isn't necessarily about `minopts` vs `optimized`, it's about platforms like ARM where HWIntrinsics aren't supported at all or cases like `Vector64` on `x86`.

In both of those scenarios, it is recognized as intrinsic (due to the `[Intrinsic]`) and as `mustExpand` (due to the recursion), but it isn't possible to actually expand the call since it isn't recognized  and so it was hitting https://github.com/dotnet/runtime/pull/36267/files#diff-b8d003a58643e5595d2920ca5993b952L4275

This was really only a problem for classes like `Vector64`/`Vector128`/`Vector256` as they are shared between all architectures (where classes like `x86.Sse` or `Arm.AdvSimd` have two copies: one which is recursive and one which throws). It looks to have only been showing for `minOpts` because we drop the dead code entirely otherwise and so we never try to expand the call.

Reply:
I added a comment elaborating as to why here: https://github.com/dotnet/runtime/pull/36267/files#diff-b8d003a58643e5595d2920ca5993b952R4501-R4517